src/components/text-editor/text-editor.d.ts

```diff
@@ -1,21 +1,34 @@
-import * as React from 'react';
+import * as React from "react";
 import { MarginSpacingProps } from "../../utils/helpers/options-helper";
+import { Editor, EditorState, ContentState } from "draft-js";
 
 export interface TextEditorProps extends MarginSpacingProps {
+  /** The maximum characters that the input will accept */
   characterLimit?: number;
+  /** The text for the editor's label */
   labelText: string;
+  /** onChange callback to control value updates */
   onChange: (event: object) => void;
   /** Additional elements to be rendered in the Editor Toolbar, e.g. Save and Cancel Button */
   toolbarElements?: React.ReactNode;
+  /** The value of the input, this is an EditorState immutable object */
   value: object;
   /** Flag to configure component as mandatory */
   required?: boolean;
+  /** Message to be displayed when there is an error */
   error?: string;
+  /** Message to be displayed when there is a warning */
   warning?: string;
+  /** Message to be displayed when there is an info */
   info?: string;
+  /** Number greater than 2 multiplied by line-height (21px) to override the default min-height of the editor */
   rows?: number;
 }
 
-declare const TextEditor: React.FunctionComponent<TextEditorProps>;
+declare function TextEditor(props: TextEditorProps & React.RefAttributes<Editor>): JSX.Element;
 
+export {
+  EditorState as TextEditorState,
+  ContentState as TextEditorContentState,
```

Comment:
Shoud we import `EditorState  from  carbon-react/lib/components/text-editor `or `import { EditorState } from "draft-js`
I ask this question because  with `import EditorState  from  carbon-react/lib/components/text-editor`  we don't have access to  
`EditorState.createWithContent `and `EditorState.createEmpty()`

Reply:
This is a type definition fix task, this type definition is just mirroring what is in the js file. I think that you should ask that question on the `carbon` slack channel.